probes/alerting/notifier/pagerduty/event_v2.go

```diff
@@ -0,0 +1,155 @@
+	CustomDetails map[string]string `json:"custom_details,omitempty"`
+}
+
+// EventV2Images is the data structure for an image in a PagerDuty event.
+// Images are optional.
+type EventV2Images struct {
+	Src  string `json:"src,omitempty"`
+	Alt  string `json:"alt,omitempty"`
+	Href string `json:"href,omitempty"`
+}
+
+// EventV2Links is the data structure for a link in a PagerDuty event.
+// Links are optional
+type EventV2Links struct {
+	Href string `json:"href,omitempty"`
+	Text string `json:"text,omitempty"`
+}
+
+// EventV2Action is the action to be performed on the event, documented as:
+// The type of event. Can be trigger, acknowledge or resolve.
+// https://developer.pagerduty.com/docs/ZG9jOjExMDI5NTgx-send-an-alert-event#event-action-behavior
+type EventV2Action string
+
+const (
+	Trigger     EventV2Action = "trigger"
+	Acknowledge               = "acknowledge"
+	Resolve                   = "resolve"
+)
+
+// EventV2Severity is the severity of the event, documented as:
+// The perceived severity of the status the event is describing with respect
+// to the affected system. This can be critical, error, warning or info.
+// https://developer.pagerduty.com/docs/ZG9jOjExMDI5NTgx-send-an-alert-event#parameters
+type EventV2Severity string
+
+const (
+	Critical EventV2Severity = "critical"
+	Error                    = "error"
+	Warning                  = "warning"
+	Info                     = "info"
+)
+
+// EventV2Response is the data structure that is returned from PagerDuty when
+// sending a EventV2 request.
+type EventV2Response struct {
+	Status   string `json:"status,omitempty"`
+	Message  string `json:"message,omitempty"`
+	DedupKey string `json:"dedup_key,omitempty"`
+}
+
+// sendEventV2 sends an event to PagerDuty using the V2 API.
+func (c *Client) sendEventV2(event *EventV2Request) (*EventV2Response, error) {
+	jsonBody, err := json.Marshal(event)
+	if err != nil {
+		return nil, err
+	}
+
+	req, err := http.NewRequest(http.MethodPost, c.hostname+"/v2/enqueue", bytes.NewBuffer(jsonBody))
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+
+	resp, err := c.httpClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	// check status code, return error if not 202
+	if resp.StatusCode != http.StatusAccepted {
+		return nil, fmt.Errorf("PagerDuty API returned status code %d", resp.StatusCode)
+	}
+
+	b, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	body := &EventV2Response{}
+	err = json.Unmarshal(b, body)
+	if err != nil {
+		return nil, err
+	}
+
+	return body, nil
+}
+
+// createEventV2Request creates a new PagerDuty event, from the alertFields that are passed
+// in from the alerting package.
+func (c *Client) createEventV2Request(alertFields map[string]string) *EventV2Request {
+	return &EventV2Request{
+		RoutingKey:  c.routingKey,
+		DedupKey:    eventV2DedupeKey(alertFields),
```

Comment:
I think we can use condition_id as a deduplicate key:
```suggestion
		DedupKey:    alertFields["condition_id"],
```
https://github.com/cloudprober/cloudprober/blob/13ac3c31be24e9baaacf80febba7a9b6eb63999c/probes/alerting/notifier/notifier.go#L54

Reply:
No problems, the reason I couple the dedupe id to the probe details rather than a unique key is it to prevent alert spam when an alert is flapping.

But note, there is also the PD alert grouping, which allows the user to group alerts based on the alert information (and will be detailed in docs that follow this PR) to prevent this situation. 

I have no issues with condition id, I just wanted to share the context behind the decision, which in hindsight should have also been a comment in the code.